src/v/cluster/controller.cc

```diff
@@ -572,9 +596,7 @@ ss::future<> controller::cluster_creation_hook(cluster_discovery& discovery) {
         co_return;
     }
 
-    if (
-      _raft0->last_visible_index() > model::offset{}
-      || _raft0->config().brokers().size() > 1) {
+    if (controller_log_exists()) {
```

Comment:
I had ideas that getting into this block means upgrading with `empty_seed_starts_cluster=false` which can be probably safeguarded against. Otherwise it's dead and should be removed.